src/lib/proof-system/zkprogram.ts

```diff
@@ -1230,17 +1363,16 @@ function computeFeatureFlags(gates: Gate[]): MlFeatureFlags {
     let flag = gateToFlag[gate.type];
     if (flag !== undefined) flags[flag] = true;
   }
+  return flags;
+}
+
+function featureFlagsToMl(flags: FeatureFlags): MlFeatureFlags {
   return [
     0,
-    MlBool(flags.rangeCheck0),
-    MlBool(flags.rangeCheck1),
-    MlBool(flags.foreignFieldAdd),
-    MlBool(flags.foreignFieldMul),
-    MlBool(flags.xor),
-    MlBool(flags.rot),
-    MlBool(flags.lookup),
-    MlBool(flags.runtimeTables),
-  ];
+    ...Object.entries(flags).map(([_, flag]) =>
+      MlBool(typeof flag === 'boolean' ? flag : false)
+    ),
+  ] as MlFeatureFlags;
```

Comment:
wait how can we just ignore the keys of the `flags` object? are we relying on a particular order?

oh I see that order is enforced by creating the full object in a separate function. that seems a bit brittle to me.

I also don't like how this refactor sacrifices type safety for a few saved lines of code.

IMO revert and adapt original to the undefined case, or at least make it independent of the order of the keys on `flags`

Reply:
yep thats right, didn't consider the ordering.. I reverted it, after all its not needed to massively change this anymore (a previous approach required a different structure)